.gitattributes

```diff
@@ -0,0 +1,18 @@
+* text=auto
+
+*.c   text eol=lf
+*.h   text eol=lf
+*.cc  text eol=lf
+*.cuh text eol=lf
+*.cu  text eol=lf
+*.py  text eol=lf
+*.txt text eol=lf
+*.R   text eol=lf
+*.scala text eol=lf
+*.java  text eol=lf
```

Comment:
Why convert to LF even on Windows?

Reply:
I think it's just a problem of picking one instead of having mixed newline symbols. I accidentally removed a carriage on the partition builder, and here is the warning from cpplint for using mixed newline symbol:
```
src/common/partition_builder.h:371:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:372:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:373:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:374:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:375:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:376:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:377:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:378:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:379:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:380:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:381:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
src/common/partition_builder.h:382:  Unexpected \r (^M) found; better to use only \n  [whitespace/newline] [1]
 ....
```